### PR TITLE
Fix class comparison by name in SocketFactory classes

### PR DESCRIPTION
**Pull request by ckelleyRH**
Use `Class.isAssignableFrom()` to determine underlying type rather than relying on string comparison.

### CODE REVIEW

**Comment by ckelleyRH** [comment]
Weird that the CI failed, it passed fine in my fork (except the Sonar job because I don't have the secret set up). I see Endi's version bump commit to master failed with the same container error so need to trawl back through the commit history to figure out when it started I guess.

**Comment by edewata** [comment]
I think it's the same cgroups issue being discussed in https://github.com/dogtagpki/jss/pull/908.

**Comment by edewata** [comment]
Could we just use the [Class.getConstructor()](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/Class.html#getConstructor(java.lang.Class...)) like this?
```
Class<?> c = Class.forName(m_packageName);
if (m_cipherSuites == null) {
    Constructor<?> m = c.getConstructor(String.class, int.class);
    return (Socket) m.newInstance(host, port);
} else {
    Constructor<?> m = c.getConstructor(String.class, int.class, m_cipherSuites.getClass());
    return (Socket) m.newInstance(host, port, m_cipherSuites);
}
```

**Comment by fmarco76** [comment]
> CI failed,

I have just merged the fix for the CI. The PR should be rebased to the master to fix the problem.

**Comment by ckelleyRH** [comment]
> 

That is nicer.

File: java-sdk/ldapjdk/src/main/java/netscape/ldap/LDAPSSLSocketWrapFactory.java [hunk 1]
```diff
@@ -181,22 +181,8 @@ public LDAPSSLSocket(String host, int port, String packageName)
...
-                    args[0] = host;
-                    args[1] = port;
-                    m_socket = (m[i].newInstance(args));
-                    return;
-                }
-            }
+            Constructor<?> m = c.getConstructor(String.class, int.class);
+            m_socket = m.newInstance(host, port);
             throw new LDAPException("No appropriate constructor in " +
               m_packageName, LDAPException.PARAM_ERROR);
```
Comment (edewata) [inline java-sdk/ldapjdk/src/main/java/netscape/ldap/LDAPSSLSocketWrapFactory.java]: I think this `throw` statement needs to be removed.

Comment (ckelleyRH) [reply java-sdk/ldapjdk/src/main/java/netscape/ldap/LDAPSSLSocketWrapFactory.java]: Good spot, I fixed the others but somehow missed this, thanks!

**Review by edewata — APPROVED** [review]
SonarCloud complains about an unused variable, but other than that it LGTM.

**Review by fmarco76 — APPROVED** [review]
LGTM

**Comment by ckelleyRH** [comment]
> SonarCloud complains about an unused variable, but other than that it LGTM.

Indeed, but it was already unused, so outside scope of this PR. I am fixing as a follow up PR as there are other issues I noticed which should be easy to clean while I'm at it.

**Comment by ckelleyRH** [comment]
Thanks @fmarco76 @edewata !

**Pull request merged**